mpqp/tools/theoretical_simulation.py

```diff
@@ -233,7 +232,7 @@ def validate_noisy_circuit(
     Returns:
         Weather our noise pipeline matches the theory or not.
     """
-    return exp_id_dist(circuit, shots, device) <= trust_int(circuit)
+    return bool(exp_id_dist(circuit, shots, device) <= trust_int(circuit))
```

Comment:
I would use also `if TYPE_CHECKING`.

Reply:
why is the bool needed ?

Reply:
Because of this error:
typeguard.TypeCheckError: the return value (numpy.bool_) is not an instance of bool

Reply:
Ok for me